packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -296,12 +296,19 @@ export default class ScriptTransformer {
     }
 
     if (!transformed.map) {
-      //Could be a potential freeze here.
-      //See: https://github.com/facebook/jest/pull/5177#discussion_r158883570
-      const inlineSourceMap = sourcemapFromSource(transformed.code);
+      try {
+        //Could be a potential freeze here.
+        //See: https://github.com/facebook/jest/pull/5177#discussion_r158883570
+        const inlineSourceMap = sourcemapFromSource(transformed.code);
 
-      if (inlineSourceMap) {
-        transformed.map = inlineSourceMap.toJSON();
+        if (inlineSourceMap) {
+          transformed.map = inlineSourceMap.toJSON();
+        }
+      } catch (e) {
+        console.warn(
+          `jest-transform: The source map produced for the file ${filename} ` +
+            'was invalid. Proceeding without source mapping for that file.',
```

Comment:
Information on what transformer is responsible may also be helpful here (for the case in which the transformer is consistently failing), but the closest I can find is the transform path, which may look a bit wonky. Is that worth adding, or should we leave the user to infer the transformer from which file is affected?

Reply:
using `_getTransformPath` seems reasonable. You can make the path to it relative from `config.rootDir`, and it shouldn't be too bad

Reply:
On further examination, the transform path is used by `require` and is thus probably a package name or relative path, so I'm including it as-is.